docs/docs/rest-api/public/api/v2/types/check.raml

```diff
@@ -0,0 +1,141 @@
+#%RAML 1.0 Library
+uses:
+  strings: stringTypes.raml
+  command: mesosCommand.raml
+  numbers: numberTypes.raml
+types:
+  HttpCheck:
+    type: object
+    properties:
+      endpoint:
+        type: strings.Name
+        description: |
+          The endpoint name to use.
+          In "host" mode checks use the hostPort. In other modes use the containerPort.
+      path?: strings.Path
+      scheme?:
+        type: strings.HttpScheme
+        default: HTTP
+  TcpCheck:
+    type: object
+    properties:
+      endpoint:
+        type: strings.Name
+        description: |
+          The endpoint name to use.
+          In "host" mode checks use the hostPort. In other modes use the containerPort.
+  CommandCheck:
+    type: object
+    properties:
+      command: command.MesosCommand
+  AppHttpCheck:
+    type: object
+    properties:
+      portIndex?:
+        type: numbers.AnyPort
+        description: |
+          Index in this app's ports array to be used for check requests.
+          An index is used so the app can use random ports,
+          like [0, 0, 0] for example, and tasks could be started with
+          port environment variables like $PORT1.
+      path?: strings.Path
+      scheme?:
+        type: strings.HttpScheme
+        default: HTTP
+  AppTcpCheck:
+    type: object
+    properties:
+      portIndex?:
+        type: numbers.AnyPort
+        description: |
+          Index in this app's ports array to be used for check requests.
+          An index is used so the app can use random ports,
+          like [0, 0, 0] for example, and tasks could be started with
+          port environment variables like $PORT1.
+
+  Check:
+    type: object
+    properties:
+      http?: HttpCheck
```

Comment:
So just I understand. We can have one of each or only one? If we can have only one, what do you think having a base type such as with [UnreachableStrategy](https://github.com/mesosphere/marathon/blob/master/docs/docs/rest-api/public/api/v2/types/unreachableStrategy.raml):

```raml
CheckType: (HttpCheck | TcpCheck | CommandCheck)
Check:
  type: object
  properties:
    check: CheckType
```

this might work

```raml
Check:
  type: object
  properties:
    check: (HttpCheck | TcpCheck | CommandCheck)
```

Reply:
if we did this for `Check` we would need to do it for `AppCheck` and it results in 
`CheckType: (HttpCheck | TcpCheck | CommandCheck)` and 
`AppCheckType: (AppHttpCheck | AppTcpCheck | CommandCheck)`

```[error] /Users/kensipe/projects/marathon/target/scala-2.12/src_managed/main/mesosphere/marathon/raml/CheckType.scala:82:12: CommandCheck is already defined as case class CommandCheck
[error] case class CommandCheck(command: MesosCommand) extends CheckType
[error]            ^
[error] one error found
```

it seems we would need to make an `AppCommandCheck` to make this work which completed defeats reuse and consistency I was hoping for.